pkg/kubectl/cmd/cp.go

```diff
@@ -301,10 +301,12 @@ func recursiveTar(srcBase, srcFile, destBase, destFile string, tw *tar.Writer) e
 		if err != nil {
 			return err
 		}
-
 		defer f.Close()
-		_, err = io.Copy(tw, f)
-		return err
+
+		if _, err := io.Copy(tw, f); err != nil {
+			return err
+		}
+		return f.Close()
```

Comment:
Duplicate `f.Close()` in L304 and L309?

Remove L304.

Reply:
No, if we get an error we still need to close.  And you must close to guarantee the file is written to disk.